src/test/coll.rs

```diff
@@ -545,3 +546,91 @@ async fn ns_not_found_suppression() {
     coll.drop(None).await.expect("drop should not fail");
     coll.drop(None).await.expect("drop should not fail");
 }
+
+async fn delete_hint_test(options: Option<DeleteOptions>, name: &str) {
+    let _guard = LOCK.run_concurrently().await;
+
+    let client = EventClient::new().await;
+    let coll = client.database(name).collection(name);
+    let _ = coll.delete_many(doc! {}, options.clone()).await;
+
+    let events = client.get_command_started_events("delete");
+    assert_eq!(events.len(), 1);
+
+    let event_hint = events.first().unwrap().command.get("hint").cloned();
+    let expected_hint = match options {
+        Some(options) => options.hint.map(|hint| hint.to_bson()),
```

Comment:
I think this match can be replaced with `options.and_then(|opts| opts.hint).map(|hint| hint.to_bson())`.